File: ios/Sources/GsCapacitorKustomerPlugin/GsCapacitorKustomerPlugin.swift
import Foundation
import Capacitor
import KustomerChat

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(GsCapacitorKustomerPlugin)
public class GsCapacitorKustomerPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "GsCapacitorKustomerPlugin"
    public let jsName = "GsCapacitorKustomer"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = GsCapacitorKustomer()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("event") ?? ""

        switch value {
            case "kustomerSignIn":
                handleKustomerSingIn(token: call.getString("jwtToken") ?? "")
            case "kustomerSignOut":
                handleKustomerSingOut()
            case "kustomerOpenChat":
                handleKustomerOpenChat()
            default:
                print("Invalid event: '\(value)'")
        }

        call.resolve([
            "value": value
        ])
    }

    func handleKustomerSingIn(token: String) {
        print("Kustomer sign in...")

        Kustomer.logIn(jwt: token) { result in
            switch result {
                case .success:
                    print("Kustomer sign in was successful")
                case .failure(let error):
                    print("Kustomer sign in error: \(error.localizedDescription)")
            }
        }
    }

    func handleKustomerSingOut() {
        print("Kustomer sign out...")

        Kustomer.logOut({ error in
            if error != nil {
                print("Kustomer sign out error: \(error?.localizedDescription ?? "")")
            }
        })
    }

    func handleKustomerOpenChat() {
        print("Kustomer show...")

        Kustomer.show()
    }
}

File: android/src/main/java/com/tavoohoh/plugin/capacitorkustomer/GsCapacitorKustomerPlugin.java
package com.tavoohoh.plugin.capacitorkustomer;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "GsCapacitorKustomer")
public class GsCapacitorKustomerPlugin extends Plugin {

    private GsCapacitorKustomer implementation = new GsCapacitorKustomer();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);

        if (value.equals("kustomerOpenChat")) {
            String email = call.getString("email");
            String jwtToken = call.getString("jwtToken");
            KustomerKt.open(email, jwtToken);
        }

        if (value.equals("kustomerSignOut")) {
            KustomerKt.logout();
        }

        JSObject ret = new JSObject();
        ret.put("value", value);
        call.resolve(ret);
    }
}
